### PR TITLE
#406 Fix MSET command consistency issue with Redis

### PR DESCRIPTION
**Pull request by ar1fKhan**
Screenshot 2024-09-16 at 10 56 06 AM

### CODE REVIEW

File: internal/eval/eval.go [hunk 1]
```diff
@@ -235,7 +235,7 @@ func evalMSET(args []string, store *dstore.Store) []byte {
 	insertMap := make(map[string]*object.Obj, len(args)/2)
 	for i := 0; i < len(args); i += 2 {
 		key, value := args[i], args[i+1]
-		oType, oEnc := deduceTypeEncoding(value)
+		oType, oEnc := object.ObjTypeString, object.ObjEncodingRaw
```
Comment (lucifercr07) [inline internal/eval/eval.go:238]: I believe the fix should be the save the val by encoding it into the associated type similar to `evalSET` operation.
Redis does the same thing trying to deduce the encoding type and save it accordingly.
```
for (j = 1; j < c->argc; j += 2) {
        c->argv[j+1] = tryObjectEncoding(c->argv[j+1]);
        setKey(c, c->db, c->argv[j], c->argv[j + 1], setkey_flags);
```

Comment (lucifercr07) [reply internal/eval/eval.go:238]: @JyotinderSingh please correct if I'm missing something here.

**Review by lucifercr07 — CHANGES_REQUESTED** [review]

File: internal/eval/eval.go (hunk 1 above)
Comment (JyotinderSingh) [reply internal/eval/eval.go]: That's correct. We should be using the exact same type deduction logic for both set and mset

File: internal/eval/eval.go
Comment (lucifercr07) [inline internal/eval/eval.go:1]: Can you please add some integration tests?

Comment (ar1fKhan) [reply internal/eval/eval.go:1]: added the integration test

**Review by lucifercr07 — APPROVED** [review]
LGTM, thanks for contributing @ar1fKhan

**Pull request merged**